packages/scandipwa/src/component/ProductCompare/ProductCompare.component.js

```diff
@@ -103,20 +103,16 @@ export class ProductCompare extends Component {
     }
 
     renderProductPrices() {
-        const { products } = this.props;
-
-        return products.map(({
-            id,
-            price_range: priceRange,
-            type_id: type,
-            dynamic_price: dynamicPrice = false
-        }) => (
+        const { products, inStock } = this.props;
+        return products.map((product) => (
+            inStock(product) ? (
             <ProductPrice
-              price={ getPrice(priceRange, dynamicPrice, {}, type) }
-              key={ id }
-              priceType={ type }
+              price={ getPrice(product.price_range, product.dynamic_price, {}, product.type_id) }
+              key={ product.id }
+              priceType={ product.type_id }
               isPreview
             />
+            ) : <div block="ProductCompareAttributeRow" elem="OutOfStock">{ __('Out of stock') }</div>
```

Comment:
Make separate function for it and more readable. (destruct props from product, return out of stock in case product not in stock through if block, cal price before putting into price prop)